### PR TITLE
Restore NOPASSWD logins after remediation

### PR DESCRIPTION
**Pull request by eemperor**

### CODE REVIEW

**Comment by eemperor** [comment]
Example of output after execution:

```
----------
          ID: uncomment_VendorSTIG-top-/etc/sudoers.d/90-cloud-init-users
    Function: file.replace
        Name: /etc/sudoers.d/90-cloud-init-users
      Result: True
     Comment: Changes were made
     Started: 14:08:20.089034
    Duration: 8.774 ms
     Changes:
              ----------
              diff:
                  ---
                  +++
                  @@ -1,4 +1,4 @@
                   # Created by cloud-init v. 19.4 on Tue, 02 Mar 2021 02:57:53 +0000

                   # User rules for maintuser
                  -#   maintuser ALL=(root) NOPASSWD:ALL
                  +   maintuser ALL=(root) NOPASSWD:ALL
----------

```

File: ash-linux/el7/VendorSTIG/remediate.sls [hunk 1]
```diff
@@ -23,3 +24,15 @@ run_{{ stig_id }}-remediate:
   cmd.run:
     - name: 'oscap xccdf eval --remediate --profile {{ scapProf }} {{ dsfile }} || true'
     - cwd: '/root'
+
+# Restore NOPASSWD remediation to sudoers.d files
+{%- for sudoer in sudoerFiles %}
+uncomment_{{ stig_id}}-{{ sudoer }}:
+  file.replace:
+    - name: {{ sudoer }}
+    - pattern: '(#)(.* NOPASSWD)'
```
Comment (lorengordon) [inline ash-linux/el7/VendorSTIG/remediate.sls:33]: @eemperor in that output, it looks like it leaves leading spaces? do we know that sudo ignores those, or do we need to remove them also?

```
-#   maintuser ALL=(root) NOPASSWD:ALL
+   maintuser ALL=(root) NOPASSWD:ALL   <==== appears to be two leading spaces?
```

```suggestion
    - pattern: '(#[ \t]*)(.* NOPASSWD)'
```

**Review by lorengordon — COMMENTED** [review]
Would also like to confirm that this works when the file contains multiple sudoer lines:

1. Multiple sudoer lines, all with commented NOPASSWD entries
2. Multiple sudoer lines, some of which are commented and some not
3. Multiple sudoer lines, none of which are commented

File: ash-linux/el7/VendorSTIG/remediate.sls (hunk 1 above)
Comment (eemperor) [reply ash-linux/el7/VendorSTIG/remediate.sls:33]: From what I'm seeing, sudo ignores the leading spaces but I'll give your suggestion a try.

**Comment by eemperor** [comment]
I can test the different scenarios using the state separately.  However, when used in conjunction with the remediation, 2 and 3 shouldn't happen right?

**Comment by lorengordon** [comment]
_shouldn't_ happen, no. but the scap remediation logic could change its mind again...

**Comment by eemperor** [comment]
True!

**Comment by eemperor** [comment]
Results for the three scenarios.  Also tested multiple files as well.

All commented:
```
local:
----------
          ID: uncomment_VendorSTIG-top-/etc/sudoers.d/90-cloud-init-users
    Function: file.replace
        Name: /etc/sudoers.d/90-cloud-init-users
      Result: True
     Comment: Changes were made
     Started: 14:45:31.873805
    Duration: 4.997 ms
     Changes:
              ----------
              diff:
                  ---
                  +++
                  @@ -1,8 +1,8 @@
                   # Created by cloud-init v. 19.4 on Tue, 02 Mar 2021 02:57:53 +0000

                   # User rules for maintuser
                  -# maintuser ALL=(root) NOPASSWD:ALL
                  +maintuser ALL=(root) NOPASSWD:ALL

                   # Test rules
                  -# ec2-user  ALL=(root) NOPASSWD:ALL
                  -# ecs-user  ALL=(root) NOPASSWD:ALL
                  +ec2-user  ALL=(root) NOPASSWD:ALL
                  +ecs-user  ALL=(root) NOPASSWD:ALL
----------
          ID: uncomment_VendorSTIG-top-/etc/sudoers.d/ssm-agent-users
    Function: file.replace
        Name: /etc/sudoers.d/ssm-agent-users
      Result: True
     Comment: Changes were made
     Started: 14:45:31.878958
    Duration: 1.853 ms
     Changes:
              ----------
              diff:
                  ---
                  +++
                  @@ -1,2 +1,2 @@
                   # User rules for ssm-user
                  -# ssm-user ALL=(ALL) NOPASSWD:ALL
                  +ssm-user ALL=(ALL) NOPASSWD:ALL

Summary for local
------------
Succeeded: 2 (changed=2)
Failed:    0
------------
Total states run:     2
Total run time:   6.850 ms
```
All uncommented:
```
local:
----------
          ID: uncomment_VendorSTIG-top-/etc/sudoers.d/90-cloud-init-users
    Function: file.replace
        Name: /etc/sudoers.d/90-cloud-init-users
      Result: True
     Comment: No changes needed to be made
     Started: 14:46:25.546315
    Duration: 3.419 ms
     Changes:
----------
          ID: uncomment_VendorSTIG-top-/etc/sudoers.d/ssm-agent-users
    Function: file.replace
        Name: /etc/sudoers.d/ssm-agent-users
      Result: True
     Comment: No changes needed to be made
     Started: 14:46:25.549890
    Duration: 1.254 ms
     Changes:

Summary for local
------------
Succeeded: 2
Failed:    0
------------
Total states run:     2
Total run time:   4.673 ms
```
Mix of commented and uncommented:
```
local:
----------
          ID: uncomment_VendorSTIG-top-/etc/sudoers.d/90-cloud-init-users
    Function: file.replace
        Name: /etc/sudoers.d/90-cloud-init-users
      Result: True
     Comment: Changes were made
     Started: 14:47:34.195645
    Duration: 4.53 ms
     Changes:
              ----------
              diff:
                  ---
                  +++
                  @@ -1,8 +1,8 @@
                   # Created by cloud-init v. 19.4 on Tue, 02 Mar 2021 02:57:53 +0000

                   # User rules for maintuser
                  -# maintuser ALL=(root) NOPASSWD:ALL
                  +maintuser ALL=(root) NOPASSWD:ALL

                   # Test rules
                   ec2-user  ALL=(root) NOPASSWD:ALL
                  -# ecs-user  ALL=(root) NOPASSWD:ALL
                  +ecs-user  ALL=(root) NOPASSWD:ALL
----------
          ID: uncomment_VendorSTIG-top-/etc/sudoers.d/ssm-agent-users
    Function: file.replace
        Name: /etc/sudoers.d/ssm-agent-users
      Result: True
     Comment: No changes needed to be made
     Started: 14:47:34.200331
    Duration: 1.421 ms
     Changes:

Summary for local
------------
Succeeded: 2 (changed=1)
Failed:    0
------------
Total states run:     2
Total run time:   5.951 ms
```

**Review by lorengordon — APPROVED** [review]
looks great! 🚀

**Pull request merged**